### PR TITLE
Add frontend authorization for procurement tracker

### PR DESCRIPTION
**Pull request by josbell**
## What changed

Added frontend authorization to the procurement tracker to prevent non-team members from editing forms. This provides clear visual feedback (disabled tab with tooltip, disabled form elements) while maintaining backend enforcement of permissions.

Implemented authorization at three levels:
1. **Tab Level** - Disabled tab in DetailsTabs with explanatory tooltip
2. **Form Level** - Parent component passes `isEditable` to all steps
3. **Component Level** - All step components disable form elements for unauthorized users

Follows existing `agreement._meta.isEditable` pattern used throughout the app for consistency.

### API Standardization

> **⚠️ IMPORTANT: This PR makes BREAKING changes to step component APIs. No backward compatibility is maintained - all prop changes are atomic within this single PR.**

Standardized the procurement tracker step component APIs to use a unified `isDisabled` prop across all four steps, eliminating inconsistent control flags and providing a single source of truth for step disabled state.

**Previous state (inconsistent):**
- Step One: `isEditable` (positive semantics)
- Step Two: `isDisabled` (negative semantics)
- Step Three: `hasActiveTracker` (positive semantics, inverted logic ❌)
- Step Four: `isDisabled` (negative semantics)

**New state (consistent):**
- **All steps: `isDisabled`** (negative semantics, matches HTML standard ✅)
- Parent calculates: `const isStepDisabled = !hasActiveTracker || !isEditable`
- All steps receive: `isDisabled={isStepDisabled}`

**Implementation approach:**
- ✨ **Clean breaking change** - All steps updated atomically in this PR
- **NO backward compatibility** - Old props (`isEditable`, `hasActiveTracker`) completely removed
- Only one caller (`AgreementProcurementTracker.jsx`) - updated in same commit
- Fixed Step 3 inverted logic bug
- Consistent with native HTML `disabled` attribute semantics

## Issue

https://github.com/HHS/OPRE-OPS/issues/5196

### Test as Non-Team Member (unauthorized user)

1. Log in as a user who is NOT a team member on an agreement
2. Navigate to an agreement detail page
3. **Verify tab is disabled:**
   - Procurement Tracker tab should be greyed out and not clickable
   - Hover over tab - tooltip should say "Only agreement team members can edit the procurement tracker"
   - Try clicking - should not navigate
4. **Verify direct URL access protection:**
   - Navigate directly to `/agreements/{id}/procurement-tracker`
   - Page should load but all form elements disabled in ALL steps:
     - Step 1: Pre-solicitation checkbox, users combo box, date picker, notes, buttons all disabled
     - Step 2: All form elements disabled
     - Step 3: Solicitation dates, checkbox, users, date picker, notes, buttons all disabled
     - Step 4: All form elements disabled
5. **Verify read-only access:**
   - Can open/close step accordions
   - Can view completed steps (if any exist)

### Test as Team Member (authorized user)

1. Log in as a user who IS a team member on an agreement
2. Navigate to the agreement detail page
3. **Verify tab is enabled:**
   - Procurement Tracker tab should be clickable
   - Click tab - should navigate successfully
4. **Verify form functionality across ALL steps:**
   - All form elements are interactive in active steps
   - Step 1: Can check checkbox, select users, enter dates, add notes, complete step
   - Step 2: All form controls work
   - Step 3: Can enter solicitation dates, check checkbox, select users, complete step
   - Step 4: All form controls work
   - Validation and step completion flow works normally

### Test as Super User

1. Log in as a super user
2. Navigate to ANY agreement (even where not a team member)
3. **Verify bypass:**
   - Procurement Tracker tab is enabled
   - All form elements are interactive in all steps
   - Can edit and save changes

### Test No Active Tracker (Edge Case)

1. Log in as a team member on an agreement WITHOUT an active procurement tracker
2. Navigate to procurement tracker tab
3. **Verify consistent disabled state:**
   - All steps show forms disabled (read-only mode)
   - Behavior is consistent across all four steps
   - Can view step structure but cannot interact with controls

### Edge Cases

- Agreement with missing `_meta` property → defaults to not editable
- Agreement with `_meta.isEditable = undefined` → defaults to not editable
- Steps in COMPLETED status → shows read-only view for all users
- No active tracker → all steps disabled consistently

## A11y impact

- [x] No accessibility-impacting changes in this PR

Authorization is conveyed through:
- Disabled state on interactive elements (native HTML `disabled` attribute)
- Tooltip with clear explanation (ARIA-compliant Tooltip component)
- No changes to keyboard navigation, focus management, or screen reader announcements

## Screenshots

_Not applicable - authorization changes don't significantly alter visual appearance beyond disabled states_

## Definition of Done Checklist
- [x] OESA: Code refactored for clarity
- [x] OESA: Dependency rules followed
- [x] Automated unit tests updated and passed (all 2097 tests passing)
- [ ] Automated integration tests updated and passed (no E2E tests for procurement tracker authorization)
- [x] Automated quality tests updated and passed (lint, format checks pass)
- [ ] Automated load tests updated and passed (N/A for frontend authorization)
- [x] Automated a11y tests updated and passed (no accessibility regressions)
- [ ] Automated security tests updated and passed (frontend only - backend enforces)
- [x] 90%+ Code coverage achieved (coverage maintained/improved)
- [ ] Form validations updated (N/A - no validation changes)

## Links

**Related Documentation:**
- Plan implementation details: See branch commit messages
- Pattern follows: `AgreementDetails.jsx` line 34 (existing `isEditable` pattern)
- API standardization: All step components use unified `isDisabled` prop

**Testing Evidence:**
- ✅ All procurement tracker tests passing (481 tests)
- ✅ Full frontend test suite passing (2097 tests)
- ✅ Lint and format checks passing
- ✅ Pre-commit hooks passing

**Recent Updates:**
- Fixed JSDoc and test naming to match current API (commit 884ae54)

**Key Benefits:**
- ✨ **Clean, simple implementation** - No backward compatibility complexity
- Unified API across all procurement tracker steps
- Eliminated inverted logic confusion in Step Three
- Single source of truth for disabled state (`isStepDisabled` in parent)
- Consistent with HTML standard semantics (`disabled` attribute)
- Easier to maintain and extend

### CODE REVIEW

File: frontend/src/components/Agreements/ProcurementTracker/ProcurementTrackerStepOne/ProcurementTrackerStepOne.test.jsx [hunk 1]
```diff
@@ -961,4 +961,206 @@ describe("ProcurementTrackerStepOne", () => {
...
+            render(
+                <ProcurementTrackerStepOne
+                    stepStatus="PENDING"
+                    stepOneData={mockStepOneData}
+                    isActiveStep={true}
+                    handleSetCompletedStepNumber={mockHandleSetCompletedStepNumber}
+                    authorizedUsers={mockAllUsers}
+                    // isEditable not provided - should default to true
+                />
+            );
```
[the remark this answers is not in the document]
Comment (josbell) [reply frontend/src/components/Agreements/ProcurementTracker/ProcurementTrackerStepOne/ProcurementTrackerStepOne.test.jsx]: **Status: Fixed** ✅

Good catch - the test comments referenced the wrong prop name.

**Fixed in commit 884ae54:**
- Updated comments from referencing `isEditable` to `isDisabled`
- Changed: "isEditable not provided - should default to true" → "isDisabled not provided - should default to undefined/false"
- Changed: "Checkbox should be enabled because isActiveStep=true and isEditable defaults to true" → "...and isDisabled defaults to false"

Comments now accurately reflect the component's current API.
